tot/models.py

```diff
@@ -225,3 +231,302 @@ def maybe_drop_added_dates(self, predicted, df):
             predicted_new, received_ID_col_pred, received_single_time_series_pred, received_dict_test_pred
         )
         return predicted, df
+
+
+@dataclass
+class SeasonalNaiveModel(Model):
+    """
+    A `SeasonalNaiveModel` is a naive model that forecasts future values of a target series based on past observations
+    of the target series of the specified period, i.e season.
+
+    The naive models act on time series data having ``fit()`` and ``predict()`` methods. The season can be specified
+    by K steps. Therefore, the model predicts the value of `K` time steps ago. When `K>1`, it repeats the last `K`
+    values.
+
+    Parameters
+    ----------
+        K : int
+            the number of last time steps to repeat
+        n_forecasts : int
+            Number of steps ahead of prediction time step to forecast.
+    Note
+    ----
+        ``Supported capabilities``
+        * univariate time series
+        * single target
+        * manually select season steps K
+        * n_forecats>1
+
+        ``Not supported capabilities``
+        * autoregression
+        * past covariates
+        * future covariates
+
+        ``Planned capabilities``
+        * auto-detect season
+        * multi-variate time series only with multi-target
+        * frequency check and optional frequency conversion
+    """
+
+    model_name: str = "SeasonalNaive"
+    model_class: Type = None  # make sense?
+
+    def __post_init__(self):
+        # no installation checks required
+
+        # expect no _data_params
+        data_params = self.params["_data_params"]
+        assert len(data_params) == 0
+        custom_seasonalities = None  # TODO: auto-detect season from metadata
+
+        # Verify expected model_prams: K > 1, n_forecasts >1
+        model_params = deepcopy(self.params)
+        model_params.pop("_data_params")
+        assert len(model_params) == 2, "Model parameters K and n_forecasts must be provided"
+        self.n_forecasts = model_params["n_forecasts"]
+        assert self.n_forecasts > 1, "Model parameter n_forecasts must be >1. "
+        self.K = model_params["K"]  # for seasonal naive K is input parameter
+        assert self.K > 1, "Model parameter K must be >1 for seasonal naive. For K=1 select NaiveModel instead. "
+        self.n_lags = None  # TODO: should not be set to None. Find different solution.
+
+    def fit(self, df: pd.DataFrame, freq: str):
+        """Fits the naive model.
+        Naive models do not need to be explicitly fitted. However, we store fitting-related information.
```

Comment:
what is stored? freq?

Reply:
@ourownstory Yes. Should we rid of the `fit()` completely?

Reply:
move to init